benchmarks/src/main/kotlin/bfschannel/BfsChannelBenchmark.kt

```diff
@@ -0,0 +1,362 @@
+/**
+ * Iterations count for each graph
+ */
+private const val ITERATIONS = 5
+/**
+ * Number of coroutines that are used to execute bfs in parallel
+ */
+private val COROUTINES = listOf(1, 4, 8, 16)
+/**
+ * Output file for the benchmark results
+ */
+private val RESULT_FILE = "plots-generators/resultsBfs.csv"
+
+/**
+ * This benchmark tests channel as a working queue, as a queue under contention.
+ * The benchmark creates or downloads graphs, then executes parallel BFS using channel as a queue, executes sequential BFS,
+ * compares the results and computes execution times.
+ *
+ * NB: this benchmark works painfully slow without syncronization on methods [kotlinx.coroutines.internal.LockFreeLinkedListNode.remove]
+ * and [kotlinx.coroutines.internal.LockFreeLinkedListNode.helpDelete] (or other channels fixes).
+ */
+fun main() {
+    val results = ArrayList<String>()
+    for (graphCreator in GRAPHS) {
+        val graphName = graphCreator.name
+        println("=== $graphName ===")
+        val graph = graphCreator.getGraph()
+
+        val startNode = graph[0]
+
+        // warmup
+        val sequentialResults = runSequentialBfs(startNode, graph)
+
+        // benchmark iterations
+        val sequentialExecutionTimes = (1..ITERATIONS).map { runSequentialBfs(startNode, graph).executionTime }
+
+        results += "$graphName,0,${sequentialExecutionTimes.average() / 1_000_000},${sequentialExecutionTimes.standardDeviation() / 1_000_000}"
+
+        println("sequential execution time = ${sequentialExecutionTimes.average() / 1_000_000}ms " +
+                "std = ${sequentialExecutionTimes.standardDeviation() / 1_000_000}ms")
+
+        for (coroutines in COROUTINES) {
+            // warmup
+            runParallelBfs(startNode, graph, coroutines)
+
+            // benchmark iterations
+            val parallelExecutionTimes = (1..ITERATIONS).map {
+                val parallelResults = runParallelBfs(startNode, graph, coroutines)
+                check(parallelResults.distances.contentEquals(sequentialResults.distances)) { "Results found using parallel and sequential bfs are not the same" }
+                parallelResults.executionTime
+            }
+
+            results += "$graphName,$coroutines,${parallelExecutionTimes.average() / 1_000_000}," +
+                    "${parallelExecutionTimes.standardDeviation() / 1_000_000}"
+            println("coroutines count = $coroutines, " +
+                    "parallel execution time = ${parallelExecutionTimes.average() / 1_000_000}ms " +
+                    "std = ${parallelExecutionTimes.standardDeviation() / 1_000_000}ms")
+        }
+    }
+    PrintWriter(RESULT_FILE).use { writer ->
+        writer.println("graphName,coroutines,executionTimeAvgMs,executionTimeStdMs")
+        for (result in results) {
+            writer.println(result)
+        }
+    }
+}
+
+private fun runParallelBfs(startNode: Node, graph: List<Node>, coroutines: Int): ExecutionResults {
```

Comment:
You have almost the same code for `runParallelBfs` and `runSequentialBfs`. Let's create an inline function which abstracts the common logic. See the following example:
```
inline fun runBfs(startNode: Node, graph: Graph, algo: () -> ExecutionResults): ExecutionResults
```